### ISSUE TITLE
note@stem.len="0" problem

### ISSUE THREAD

**Issue by craigsapp** [rism-ch/verovio#1715]
Notes with a stem length of zero are displayed correctly for half notes and quarter notes, but not for notes shorter than a quarter note .  Instead, there are negative stems when there should be no stems:

<img width="1031" alt="Screen Shot 2020-09-30 at 10 45 30 AM" src="https://user-images.githubusercontent.com/3487289/94721075-334a5d80-030a-11eb-808b-f159823fdfd0.png">

Notes marked in red are incorrect because the stem moves to a negative value for non-negative stem lengths.

* The first measure has stems with length zero.  All notes less than a quarter note have stems that are on the wrong side of the note (see measures 6 and 7).  A length of zero is a special case where the stems should be made invisible (with no flags as is being done correctly here).
* The second measure has default stem lengths.  Rendered as expected.
* Measure 3 has lengths of 12.  I am wondering about this case: I would expect all  stem lengths to be the same, since there is enough space for all flags on the 256th note when the stem is this long.
* Measure 4 has lengths of 5.  Rendered as expected.
* Measure 5 has lengths of 3.  Rendered as expected.
* Measure 6 has lengths of 2.  Stems on all notes shorter than an eighth note pop to the other side of the note, which should be disallowed (place top flag adjacent to notehead below note for stem down).
* Measure 7 has lengths of 1.  Stems on all notes shorter than a quarter note pop to other side of note, which should be disallowed.

MEI test data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
 <meiHead>
  <fileDesc>
   <titleStmt>
    <title />
   </titleStmt>
   <pubStmt />
  </fileDesc>
  <encodingDesc>
   <appInfo>
    <application isodate="2020-09-30T10:31:53" version="3.0.0-dev-069b8f8-dirty">
     <name>Verovio</name>
     <p>Transcoded from Humdrum</p>
    </application>
   </appInfo>
  </encodingDesc>
  <workList>
   <work>
    <title />
   </work>
  </workList>
 </meiHead>
 <music>
  <body>
   <mdiv xml:id="mdiv-0000000883546152">
    <score xml:id="score-0000002058241306">
     <scoreDef xml:id="scoredef-0000000413398771">
      <staffGrp xml:id="staffgrp-0000000954275221">
       <staffDef xml:id="staffdef-0000001838550518" n="1" lines="5">
        <clef xml:id="clef-0000000681982180" shape="G" line="2" />
       </staffDef>
      </staffGrp>
     </scoreDef>
     <section xml:id="section-L1F1">
      <measure xml:id="measure-L1" n="1">
       <staff xml:id="staff-0000001806169646" n="1">
        <layer xml:id="layer-L1F1N1" n="1">
         <note xml:id="note-L4F1" dur="2" oct="4" pname="b" stem.len="0.000000" />
         <note xml:id="note-L6F1" dur="4" oct="4" pname="b" stem.len="0.000000" />
         <note xml:id="note-L8F1" dur="8" oct="4" pname="b" color="red" stem.len="0.000000" />
         <note xml:id="note-L10F1" dur="16" oct="4" pname="b" color="red" stem.len="0.000000" />
         <note xml:id="note-L12F1" dur="32" oct="4" pname="b" color="red" stem.len="0.000000" />
         <note xml:id="note-L14F1" dur="64" oct="4" pname="b" color="red" stem.len="0.000000" />
         <note xml:id="note-L16F1" dur="128" oct="4" pname="b" color="red" stem.len="0.000000" />
         <note xml:id="note-L18F1" dur="256" oct="4" pname="b" color="red" stem.len="0.000000" />
        </layer>
       </staff>
      </measure>
      <measure xml:id="measure-L19" n="2">
       <staff xml:id="staff-L19F1N1" n="1">
        <layer xml:id="layer-L19F1N1" n="1">
         <note xml:id="note-L20F1" dur="2" oct="4" pname="b" />
         <note xml:id="note-L21F1" dur="4" oct="4" pname="b" />
         <note xml:id="note-L22F1" dur="8" oct="4" pname="b" />
         <note xml:id="note-L23F1" dur="16" oct="4" pname="b" />
         <note xml:id="note-L24F1" dur="32" oct="4" pname="b" />
         <note xml:id="note-L25F1" dur="64" oct="4" pname="b" />
         <note xml:id="note-L26F1" dur="128" oct="4" pname="b" />
         <note xml:id="note-L27F1" dur="256" oct="4" pname="b" />
        </layer>
       </staff>
      </measure>
      <measure xml:id="measure-L18" n="3">
       <staff xml:id="staff-L18F1N1" n="1">
        <layer xml:id="layer-L89F1N1" n="1">
         <note xml:id="note-L29F1" dur="2" oct="4" pname="b" stem.len="12.000000" />
         <note xml:id="note-L30F1" dur="4" oct="4" pname="b" stem.len="12.000000" />
         <note xml:id="note-L31F1" dur="8" oct="4" pname="b" stem.len="12.000000" />
         <note xml:id="note-L32F1" dur="16" oct="4" pname="b" stem.len="12.000000" />
         <note xml:id="note-L33F1" dur="32" oct="4" pname="b" stem.len="12.000000" />
         <note xml:id="note-L34F1" dur="64" oct="4" pname="b" stem.len="12.000000" />
         <note xml:id="note-L35F1" dur="128" oct="4" pname="b" stem.len="12.000000" />
         <note xml:id="note-L36F1" dur="256" oct="4" pname="b" stem.len="12.000000" />
        </layer>
       </staff>
      </measure>
      <measure xml:id="measure-L37" n="4">
       <staff xml:id="staff-L37F1N1" n="1">
        <layer xml:id="layer-L37F1N1" n="1">
         <note xml:id="note-L38F1" dur="2" oct="4" pname="b" stem.len="5.000000" />
         <note xml:id="note-L39F1" dur="4" oct="4" pname="b" stem.len="5.000000" />
         <note xml:id="note-L40F1" dur="8" oct="4" pname="b" stem.len="5.000000" />
         <note xml:id="note-L41F1" dur="16" oct="4" pname="b" stem.len="5.000000" />
         <note xml:id="note-L42F1" dur="32" oct="4" pname="b" stem.len="5.000000" />
         <note xml:id="note-L43F1" dur="64" oct="4" pname="b" stem.len="5.000000" />
         <note xml:id="note-L44F1" dur="128" oct="4" pname="b" stem.len="5.000000" />
         <note xml:id="note-L45F1" dur="256" oct="4" pname="b" stem.len="5.000000" />
        </layer>
       </staff>
      </measure>
      <measure xml:id="measure-L37" n="5">
       <staff xml:id="staff-L46F1N1" n="1">
        <layer xml:id="layer-L46F1N1" n="1">
         <note xml:id="note-L47F1" dur="2" oct="4" pname="b" stem.len="3.000000" />
         <note xml:id="note-L48F1" dur="4" oct="4" pname="b" stem.len="3.000000" />
         <note xml:id="note-L49F1" dur="8" oct="4" pname="b" stem.len="3.000000" />
         <note xml:id="note-L50F1" dur="16" oct="4" pname="b" stem.len="3.000000" />
         <note xml:id="note-L51F1" dur="32" oct="4" pname="b" stem.len="3.000000" />
         <note xml:id="note-L52F1" dur="64" oct="4" pname="b" stem.len="3.000000" />
         <note xml:id="note-L53F1" dur="128" oct="4" pname="b" stem.len="3.000000" />
         <note xml:id="note-L54F1" dur="256" oct="4" pname="b" stem.len="3.000000" />
        </layer>
       </staff>
      </measure>
      <measure xml:id="measure-L55" n="6">
       <staff xml:id="staff-L55F1N1" n="1">
        <layer xml:id="layer-L55F1N1" n="1">
         <note xml:id="note-L56F1" dur="2" oct="4" pname="b" stem.len="2.000000" />
         <note xml:id="note-L57F1" dur="4" oct="4" pname="b" stem.len="2.000000" />
         <note xml:id="note-L58F1" dur="8" oct="4" pname="b" stem.len="2.000000" />
         <note xml:id="note-L59F1" dur="16" oct="4" pname="b" stem.len="2.000000" color="red" />
         <note xml:id="note-L60F1" dur="32" oct="4" pname="b" stem.len="2.000000" color="red" />
         <note xml:id="note-L61F1" dur="64" oct="4" pname="b" stem.len="2.000000" color="red" />
         <note xml:id="note-L62F1" dur="128" oct="4" pname="b" stem.len="2.000000" color="red" />
         <note xml:id="note-L63F1" dur="256" oct="4" pname="b" stem.len="2.000000" color="red" />
        </layer>
       </staff>
      </measure>
      <measure xml:id="measure-L64" right="end" n="7">
       <staff xml:id="staff-L64F1N1" n="1">
        <layer xml:id="layer-L64F1N1" n="1">
         <note xml:id="note-L65F1" dur="2" oct="4" pname="b" stem.len="1.000000" />
         <note xml:id="note-L66F1" dur="4" oct="4" pname="b" stem.len="1.000000" />
         <note xml:id="note-L67F1" dur="8" oct="4" pname="b" stem.len="1.000000" color="red" />
         <note xml:id="note-L68F1" dur="16" oct="4" pname="b" stem.len="1.000000" color="red" />
         <note xml:id="note-L69F1" dur="32" oct="4" pname="b" stem.len="1.000000" color="red" />
         <note xml:id="note-L70F1" dur="64" oct="4" pname="b" stem.len="1.000000" color="red" />
         <note xml:id="note-L71F1" dur="128" oct="4" pname="b" stem.len="1.000000" color="red" />
         <note xml:id="note-L72F1" dur="256" oct="4" pname="b" stem.len="1.000000" color="red" />
        </layer>
       </staff>
      </measure>
     </section>
    </score>
   </mdiv>
  </body>
 </music>
</mei>
```

**Comment by rettinghaus** [comment]
Should a stem length of zero actually still render flags?

**Comment by craigsapp** [comment]
> Should a stem length of zero actually still render flags?

`@stem.len="0"` is the current method for not displaying a stem on a note/chord.  See issue https://github.com/rism-ch/verovio/issues/375 for discussion about how to represent a stemless note.  If there is an alternate and/or more proper way of doing it in MEI, then whether or not to display the flag will not be important.  But if there is no alternative way of suppressing the stem, then `@stem.len="0"` should not display the flags.

**Comment by rettinghaus** [comment]
@craigsapp you are wrong. The current method for not displaying a stem on a note/chord would be `stem.visible="false"`. See issue #375.

**Comment by craigsapp** [comment]
That is great! 

Since I am creator of the request for `@stem.visible`, I cannot possibly be wrong, but rather I am misinformed.  Is there also now a way to make invisible ties as discussed in issue https://github.com/music-encoding/music-encoding/issues/469?

-------------

So `@stem.len="0"` should still display the flags, and it would look similar to `@stem.len="1"` and `@stem.len="2"`, where the flags are touching the notehead, something like:

<img width="214" alt="Screen Shot 2020-10-04 at 6 42 56 AM" src="https://user-images.githubusercontent.com/3487289/95017266-2cbf2d00-060d-11eb-82ec-0b83b62dd186.png">

but slightly shorter.  But this becomes an extreme corner case, and not very important (other than to remove the negative stems to prevent artifacts).

**Comment by craigsapp** [comment]
This is fixed by https://github.com/rism-ch/verovio/pull/1719/commits/858b3b1dbdecacf3c1c68b1172058317daf6b30b @rettinghaus ?

**Comment by rettinghaus** [comment]
Yes, fixed by 858b3b1:
![stemlen](https://user-images.githubusercontent.com/7693447/95019252-8737a800-0664-11eb-8bf2-50b0d981a845.png)

**Comment by craigsapp** [comment]
Looks good &mdash; not perfect, but close enough for now, since these cases should be very rare in real music.
